client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+    useState,
+    useEffect 
+} from "react";
+import {useNavigate} from "react-router-dom";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText"
+import DialogActions from "@mui/material/DialogActions";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Link from "@mui/material/Link";
+import Tooltip from "@mui/material/Tooltip";
+import Fab from "@mui/material/Fab";
+import SendIcon from "@mui/icons-material/Send";
+import { 
+    usePostZoopMutation, 
+    useGetAllUsersQuery,
+    useGetMeQuery
+}  from "../features/api";
+import { useSelector } from "react-redux";
+import { RootState} from "../app/store";
+import type { User } from '../../src/types/custom'
+
+const SendZoopDialog = () => {
+    const [open, setOpen] = useState(false);
+    const [username, setUsername] = useState("");
+    const [content, setContent] = useState("");
+    const [zoopSent, setZoopSent] = useState(false);
+    const [lastZoopSentId, setLastZoopSentId] = useState<number | null>(null);
+
+    // Variable used for authorization
+    const token = useSelector((state: RootState) => state.auth.token);
+
+    const currentUser = useGetMeQuery().data;
+
+    const [postZoop, { isLoading: isPostZoopLoading, isError, data: zoopData }] = usePostZoopMutation();
+    const {data: usersData, isLoading: isGetAllUsersLoading, error } = useGetAllUsersQuery() 
+
+    const navigate = useNavigate();
+
+    const handleSendZoopClick = () => {
+        if (!usersData) {
+            throw new Error('No users to send Zoop to');
+            // TODO: How best to render a message about this error in UI?
+        } else {
+            const [recipient] = usersData.users.filter(user => user.username === username);
```

Comment:
You should not need to get the user object - you should have the user stored in state, not just the username

Reply:
I was using this logic to make sure that the correct user is selected based on the username that is being asked for in the send Zoop dialog. Not sure how else to do this. How should I have done this differently?

Reply:
When you say I don't need  to "get the user" object, I take that to mean that I don't need to be calling filter on the array of users. But this was my way of getting the selected user's ID to put in the Zoop. There is no other place with the users' IDs except the user objects in this array. When you say, "the user stored in state, not just the username" are you wanting me to store a user object in state with ll of the other properties? If so, I would need to query to get it as it is not in the redux store.

Reply:
Although now that I'm looking at the Autocomplete component I set up, the ID in the input object (`{label: string, id: number}`) should be the same id that is in the user's object in the database because it mapped through all the users to create the objects for Autocomplete. I was thinking that the IDs wouldn't match up. I think I see what you are saying now.

Reply:
You're talking about the selected user / recipient, right? What do you mean by "get the user object"? Do you mean find the selected user using a filter of `usersData`?

Reply:
I meant you shouldn't need to filter to find which user was selected after they select it (that means looping through the entire list of all users before you can submit), you should instead select the user in state by ID with the Autocomplete, and then you will have that user's ID stored in state.